### PR TITLE
Flares for osquery

### PR DESCRIPTION
**Pull request by RebeccaMahany**
* Runs osqueryd --version and times how long it takes
* Runs launcher --interactive with a single query and times how long it takes

### CODE REVIEW

File: pkg/debug/checkups/osquery.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	out, err := cmd.CombinedOutput()
+	o.executionTimes[strings.Join(cmd.Args, " ")] = fmt.Sprintf("%d ms", time.Now().UnixMilli()-startTime)
+	if err != nil {
+		return "", fmt.Errorf("running %s version: err %w, output %s", osquery, err, string(out))
+	}
+
+	return strings.TrimSpace(string(out)), nil
+}
+
+func (o *osqueryCheckup) interactive(ctx context.Context) error {
```
Comment (James-Pickett) [inline pkg/debug/checkups/osquery.go:73]: this is clever! using interactive to test no only osquery, but also that launcher and osquery can communicate

**Review by James-Pickett — APPROVED** [review]
nice

File: pkg/debug/checkups/osquery.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+func (o *osqueryCheckup) Name() string {
+	return "Osquery"
+}
+
+func (o *osqueryCheckup) Run(ctx context.Context, extraWriter io.Writer) error {
+	// Determine passing status by running osqueryd --version
+	o.executionTimes = make(map[string]string)
+	if osqueryVersion, err := o.version(ctx); err != nil {
+		o.status = Failing
+		o.summary = err.Error()
```
Comment (directionless) [inline pkg/debug/checkups/osquery.go:29]: Not needed with the error return. but okay

File: pkg/debug/checkups/osquery.go (hunk 1 above)
Comment (directionless) [reply pkg/debug/checkups/osquery.go:73]: haha. I'd assumed using the underlying libraries, but this is pretty clever.

File: pkg/debug/checkups/osquery.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+func (o *osqueryCheckup) version(ctx context.Context) (string, error) {
+	osquery := osquerydPath()
+
+	cmdCtx, cmdCancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cmdCancel()
+
+	cmd := exec.CommandContext(cmdCtx, osquery, "--version")
+	startTime := time.Now().UnixMilli()
+	out, err := cmd.CombinedOutput()
+	o.executionTimes[strings.Join(cmd.Args, " ")] = fmt.Sprintf("%d ms", time.Now().UnixMilli()-startTime)
```
Comment (directionless) [inline pkg/debug/checkups/osquery.go:65]: `cmd.String()` maybe?

File: pkg/debug/checkups/osquery.go [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+		return fmt.Errorf("creating zip file for stderr: %w", err)
+	}
+	cmd := exec.CommandContext(cmdCtx, osquery, "--ephemeral", "--disable_database", "--disable_logging", "--verbose")
+	cmd.Stderr = out
+
+	if err := cmd.Start(); err != nil {
+		return fmt.Errorf("starting osqueryd in foreground: err %w", err)
+	}
+
+	time.Sleep(runInForegroundDuration)
```
Comment (directionless) [inline pkg/debug/checkups/osquery.go:117]: So this blocks for 10s? That doesn't seem great.

File: pkg/debug/checkups/osquery.go [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+	if extraWriter == io.Discard {
+		return nil
+	}
+
+	zipWriter := zip.NewWriter(extraWriter)
+	defer zipWriter.Close()
+
+	if err := o.foreground(ctx, zipWriter); err != nil {
+		return fmt.Errorf("running osqueryd in foreground: %w", err)
+	}
```
Comment (directionless) [inline pkg/debug/checkups/osquery.go:51]: I'm not sure this adds much over the other things

File: pkg/debug/checkups/osquery.go [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+		return fmt.Errorf("running launcher interactive: %w", err)
+	}
+
+	// If we're running doctor and not flare, that's all we need
+	if extraWriter == io.Discard {
+		return nil
+	}
+
+	zipWriter := zip.NewWriter(extraWriter)
+	defer zipWriter.Close()
```
Comment (directionless) [inline pkg/debug/checkups/osquery.go:47]: I don't know if you need this at all. Though you add it to capture stdout on earlier two commands

File: pkg/debug/checkups/osquery.go (hunk 5 above)
Comment (RebeccaMahany) [reply pkg/debug/checkups/osquery.go:51]: I wasn't sure either, esp given the blocking for 10s -- I'll take it back out

**Review by James-Pickett — APPROVED** [review]

File: pkg/debug/checkups/helpers_windows.go [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+package checkups
+
+import (
+	"os/exec"
+	"syscall"
+)
+
+func hideWindow(cmd *exec.Cmd) {
+	cmd.SysProcAttr = &syscall.SysProcAttr{HideWindow: true} // prevents spawning window
```
Comment (directionless) [inline pkg/debug/checkups/helpers_windows.go:12]: neat!

**Review by directionless — APPROVED** [review]

**Pull request merged**